changelog/2836.bug

```diff
@@ -0,0 +1,3 @@
+Attempted fix for https://github.com/pytest-dev/pytest/issues/2836
```

Comment:
i propose `match fixture paths against actual path segments in order to avoid matching folders which share a prefix`

CC @nicoddemus